pyannote/core/segment.py

```diff
@@ -170,7 +170,7 @@ def __bool__(self):
 
     def __post_init__(self):
         """Round start and end up to SEGMENT_PRECISION precision (when required)"""
-        if AUTO_ROUND_TIME:
+        if self and AUTO_ROUND_TIME:
```

Comment:
As for the original `Segment.set_precision` PR, I am concerned about speed.
This should avoid unnecessary calls to `__bool__` when `AUTO_ROUND_TIME` is already `False`: 
```suggestion
        if AUTO_ROUND_TIME and self:
```

Can you please double check that it is actually faster?

Reply:
great idea